src/main/java/net/dv8tion/jda/core/entities/MessageChannel.java

```diff
@@ -466,8 +473,15 @@ protected void handleResponse(Response response, Request request)
 
         if (message != null)
         {
-            body.field("content", message.getRawContent());
-            body.field("tts", message.isTTS());
+            if (message.getEmbeds().isEmpty())
+            {
+                body.field("content", message.getRawContent());
+                body.field("tts", message.isTTS());
+            }
+            else
+            {
+                body.field("payload_json", ((MessageImpl) message).toJSONObject().toString());
```

Comment:
If we can switch completely then we should. The reason behind that being: If anything new is ever added to Message, it will be represented in the JSONObject returned by `toJSONObject()`, and will automatically be handled here with no code change. Where as if we use `body.field` we have to modify those fields in the future.

As such, lets change all file upload situations to use `payload_json`